grails-app/controllers/org/pih/warehouse/user/UserController.groovy

```diff
@@ -302,40 +302,25 @@ class UserController {
     def updateAuthUserLocale = {
 
         log.info "update auth user locale " + params
-        log.info params.locale == 'debug'
-        if (params.locale == 'debug') {
-            session.useDebugLocale = true
-        } else {
-            session.useDebugLocale = false
-            // if no locale specified, do nothing
-            if (!params.locale) {
-                redirect(controller: "dashboard", action: "index")
-            }
-
-            // convert the passed locale parameter to an actual locale
-            Locale locale = new Locale(params.locale)
+        // if no locale specified, do nothing
+        if (!params.locale) {
+            redirect(controller: "dashboard", action: "index")
+        }
```

Comment:
I know this isn't code you wrote, but we should probably remove the redirects since that's handled at the end. And instead of handling the 
```
    if (params.locale) { 
        // convert the passed locale parameter to an actual locale
        Locale locale = new Locale(params.locale)
    
        if (locale) { 
            // Change the locale
            session.locale = locale
        
            // remove all redirects since they aren't going to do anything 
        
            // Check if chosen locale is equal to translation mode locale, if so, enable localization mode
            session.useDebugLocale = localizationService.isLocalizationModeEnabled() 
        }
    }

    log.info "Redirecting to " + params?.targetUri
    if (params?.targetUri) {
        redirect(uri: params.targetUri)
        return
    }
    // redirect to the dashboard
    redirect(controller: "dashboard", action: "index")
```

LocalizationService
```
isLocalizationModeEnabled() { 
    def localizationModeLocale = new Locale(grailsApplication.config.openboxes.locale.translationModeLocale)
    return currentLocale == localizationModeLocale
```